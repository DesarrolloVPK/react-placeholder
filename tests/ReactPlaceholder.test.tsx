import * as React from 'react';
import { execSync } from 'child_process';
import * as fs from 'fs';
import * as path from 'path';
import { flatten } from 'lodash';
import ReactPlaceholder from '../src/ReactPlaceholder';
import { shallow } from 'enzyme';

jest.useFakeTimers();

describe('ReactPlaceholder', () => {

  it('renders the placeholder when the content is not ready', () => {
    const content = <div>Some content still loading...</div>;
    const tree = shallow(
      <ReactPlaceholder type='text' rows={2} ready={false}>
        {content}
      </ReactPlaceholder>
    );
    expect(tree.contains(content)).toBe(false);
    expect(tree.getElements()).toMatchSnapshot();
  });

  it('renders the placeholder only after the specified delay', () => {
    const content = <div>Some content still loading...</div>;
    const tree = shallow(
      <ReactPlaceholder
        ready
        type='text'
        rows={2}
        delay={10000}
      >
        {content}
      </ReactPlaceholder>
    );
    expect(tree.contains(content)).toBe(true);
    tree.setProps({ ready: false });
    tree.update();
    expect(tree.contains(content)).toBe(true);
    jest.runAllTimers();
    tree.update();
    expect(tree.contains(content)).toBe(false);
    expect(tree.getElements()).toMatchSnapshot();
  });

  it('renders the content when it\'s ready', () => {
    const content = <div>Some ready content</div>;
    const tree = shallow(
      <ReactPlaceholder type='text' rows={2} ready>
        {content}
      </ReactPlaceholder>
    );
    expect(tree.contains(content)).toBe(true);
    expect(tree.getElements()).toMatchSnapshot();
  });

});

describe('build', () => {

  it('build script generates every needed file', () => {
    execSync('npm run build', { stdio:[] })

    const libPath = path.resolve(__dirname, '../lib');

    const libFiles = fs.readdirSync(libPath);

    const files = flatten(libFiles.map(fileInRoot => {
      const filePath = path.resolve(libPath, fileInRoot);
      if (fs.lstatSync(filePath).isDirectory()) {
        return fs.readdirSync(filePath).map(fileInFolder => `${fileInRoot}/${fileInFolder}`);
      }
      return fileInRoot;
    }));

    expect(files).toMatchSnapshot();
  })

})
