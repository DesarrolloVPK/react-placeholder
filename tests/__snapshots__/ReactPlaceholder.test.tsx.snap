// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactPlaceholder renders a custom placeholder when the content is not ready 1`] = `
Array [
  <div
    className=""
  >
    Custom Placeholder
  </div>,
]
`;

exports[`ReactPlaceholder renders the content when it's ready 1`] = `
Array [
  <div>
    Some ready content
  </div>,
]
`;

exports[`ReactPlaceholder renders the media placeholder when the content is not ready 1`] = `
Array [
  <MediaBlock
    color="#CDCDCD"
    delay={0}
    rows={2}
  />,
]
`;

exports[`ReactPlaceholder renders the placeholder only after the specified delay 1`] = `
Array [
  <TextBlock
    color="#CDCDCD"
    delay={10000}
    rows={2}
    widths={
      Array [
        97,
        100,
        94,
        90,
        98,
        95,
        98,
        40,
      ]
    }
  />,
]
`;

exports[`ReactPlaceholder renders the rect placeholder when the content is not ready 1`] = `
Array [
  <RectShape
    color="#CDCDCD"
    delay={0}
  />,
]
`;

exports[`ReactPlaceholder renders the round placeholder when the content is not ready 1`] = `
Array [
  <RoundShape
    color="#CDCDCD"
    delay={0}
  />,
]
`;

exports[`ReactPlaceholder renders the text placeholder when the content is not ready 1`] = `
Array [
  <TextBlock
    color="#CDCDCD"
    delay={0}
    rows={2}
    widths={
      Array [
        97,
        100,
        94,
        90,
        98,
        95,
        98,
        40,
      ]
    }
  />,
]
`;

exports[`ReactPlaceholder renders the textRow placeholder when the content is not ready 1`] = `
Array [
  <TextRow
    color="#CDCDCD"
    delay={0}
    lineSpacing="0.7em"
  />,
]
`;

exports[`build build script generates every needed file 1`] = `
Array [
  "ReactPlaceholder.d.ts",
  "ReactPlaceholder.js",
  "index.d.ts",
  "index.js",
  "placeholders/MediaBlock.d.ts",
  "placeholders/MediaBlock.js",
  "placeholders/RectShape.d.ts",
  "placeholders/RectShape.js",
  "placeholders/RoundShape.d.ts",
  "placeholders/RoundShape.js",
  "placeholders/TextBlock.d.ts",
  "placeholders/TextBlock.js",
  "placeholders/TextRow.d.ts",
  "placeholders/TextRow.js",
  "placeholders/index.d.ts",
  "placeholders/index.js",
  "reactPlaceholder.css",
]
`;
